### ISSUE TITLE
Bot should not try to merge main into hotfix branches

### ISSUE THREAD

**Issue by heaths** [Azure/azure-sdk-tools#5150]
After releasing a patch per [guidelines](https://azure.github.io/azure-sdk/policies_repobranching.html#hotfix-branches), [a bot tried to merge `main` back in](https://github.com/Azure/azure-sdk-for-net/pull/33486). By either course of action laid out by the hotfix guidelines, this should never happen. In fact, seems to me for a `hotfix/`, `feature/`, or `release/` branch there should never be an automatic PR to merge *from* `main`.

**Comment by kurtzeborn** [comment]
Agreed on `hotfix/` for sure (possibly for the others).  That said, the workaround is simple... don't merge the PRs.

**Comment by heaths** [comment]
Sounds rife for mishaps for someone that doesn't know better. Seems better to never create them in the first place. I didn't merge - and opened this issue - because I know better. With more partners managing their own SDKs, not everyone may know not to merge.

**Comment by weshaggard** [comment]
Interesting I wouldn't have expected it to try and merge main that actually seems like a bug. However I also agree we probably shouldn't even have the pipeline submit these version increments outside of a long lived branch, or perhaps just main.

**Comment by weshaggard** [comment]
I don't think this is likely worth any effort to do. If we see any negative issues caused by this then we can reconsider opening another issue in the future.
